unittests/vote_processor_tests.cpp

```diff
@@ -56,7 +56,7 @@ auto create_genesis_block_state() { // block 2
    genesis->block = block;
    genesis->activated_protocol_features = std::make_shared<protocol_feature_activation_set>();
    genesis->active_finalizer_policy = std::make_shared<finalizer_policy>(new_finalizer_policy);
-   genesis->block->previous = make_block_id(1);
+   std::const_pointer_cast<signed_block>(genesis->block)->previous = make_block_id(1);
```

Comment:
```suggestion
   block->previous = make_block_id(1);
```

Reply:
This one is block_state->block so it is needed.

Reply:
I don't think so, `block` is `auto block = std::make_shared<signed_block>();`

It becomes a `signed_block_ptr` when the shared_ptr is copied to `genesis`, but `block` is still mutable as line 40 demonstrates.

Reply:
This is `genesis->block`. so `block_state::block` which is a `signed_block_ptr`

```
/home/heifner/ext/leap/unittests/vote_processor_tests.cpp:59:29: error: no viable overloaded '='
   genesis->block->previous = make_block_id(1);
   ~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~
/home/heifner/ext/leap/libraries/libfc/include/fc/crypto/sha256.hpp:13:7: note: candidate function (the implicit copy assignment operator) not viable: 'this' argument has type 'const block_id_type' (aka 'const fc::sha256'), but method is not marked const
class sha256 
      ^
/home/heifner/ext/leap/libraries/libfc/include/fc/crypto/sha256.hpp:13:7: note: candidate function (the implicit move assignment operator) not viable: 'this' argument has type 'const block_id_type' (aka 'const fc::sha256'), but method is not marked const

```

Reply:
.. but can do it before the assignment.

Reply:
Which is what you meant all along probably. Sorry.

Reply:
> genesis->block->previous = make_block_id(1);

That was not my suggestion :-)